lib/segment/src/index/field_index/map_index/mmap_map_index.rs

```diff
@@ -0,0 +1,226 @@
+            },
+        )?;
+
+        MmapHashMap::create(
+            &hashmap_path,
+            values_to_points
+                .iter()
+                .map(|(value, ids)| (<N as MapIndexKey>::from_owned(value), ids.iter().copied())),
+        )?;
+
+        MmapPointToValues::<N>::from_iter(
+            path,
+            point_to_values.iter().enumerate().map(|(idx, values)| {
+                (
+                    idx as PointOffsetType,
+                    values
+                        .iter()
+                        .map(|value| N::into_referenced(N::from_owned(value))),
+                )
+            }),
+        )?;
+
+        {
+            const BITS_IN_BYTE: usize = 8;
+            let deleted_flags_count = point_to_values.len();
+            let deleted_file = create_and_ensure_length(
+                &deleted_path,
+                BITS_IN_BYTE
+                    * BITS_IN_BYTE
+                    * deleted_flags_count.div_ceil(BITS_IN_BYTE * BITS_IN_BYTE),
+            )?;
+            let mut deleted_mmap = unsafe { MmapMut::map_mut(&deleted_file)? };
+            deleted_mmap.fill(0);
+            let mut deleted_bitflags = MmapBitSlice::from(deleted_mmap, 0);
+            for (idx, values) in point_to_values.iter().enumerate() {
+                if values.is_empty() {
+                    deleted_bitflags.set(idx, true);
+                }
+            }
+        }
+
+        Self::load(path)
+    }
+
+    pub fn flusher(&self) -> Flusher {
+        self.deleted.flusher()
+    }
+
+    pub fn clear(self) -> OperationResult<()> {
+        let files = self.files();
+        drop(self);
+        for file in files {
+            std::fs::remove_file(file)?;
+        }
+        Ok(())
+    }
+
+    pub fn files(&self) -> Vec<PathBuf> {
+        let mut files = vec![
+            self.path.join(HASHMAP_PATH),
+            self.path.join(DELETED_PATH),
+            self.path.join(CONFIG_PATH),
+        ];
+        files.extend(self.point_to_values.files());
+        files
+    }
+
+    pub fn remove_point(&mut self, idx: PointOffsetType) {
+        let idx = idx as usize;
+        if idx < self.deleted.len() && !self.deleted.get(idx).unwrap_or(true) {
+            self.deleted.set(idx, true);
+            self.deleted_count += 1;
+        }
+    }
+
+    pub fn check_values_any(&self, idx: PointOffsetType, check_fn: impl Fn(&N) -> bool) -> bool {
+        if self.deleted.get(idx as usize).unwrap_or(true) {
+            self.point_to_values
+                .check_values_any(idx, |v| check_fn(N::from_referenced(&v)))
+        } else {
+            false
+        }
+    }
+
+    pub fn get_values(
+        &self,
+        idx: PointOffsetType,
+    ) -> Option<Box<dyn Iterator<Item = N::Referenced<'_>> + '_>> {
+        if !self.deleted.get(idx as usize).unwrap_or(true) {
+            Some(Box::new(self.point_to_values.get_values(idx)?))
```

Comment:
Style nit: you could use `bool_value.then(|| …)` instead of `if bool_value { Some(…) } else { None }`.

Reply:
Refactored to avoid `if`